### PR TITLE
ENT-4057: Fixed possible segfaults in hashing functions

### PR DESCRIPTION
**Pull request by olehermanse**

### CODE REVIEW

**Comment by cf-bottom** [comment]
Thank you for submitting a PR! Maybe @vpodzime can review this?

**Comment by olehermanse** [comment]
@cf-bottom Please test this in jenkins, with exotics.

**Comment by cf-bottom** [comment]
Alright, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/1107/)

(with exotics)

https://ci.cfengine.com/job/pr-pipeline/1107/

**Comment by cf-bottom** [comment]
Alright, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/1108/)

(with exotics)

https://ci.cfengine.com/job/pr-pipeline/1108/

**Comment by cf-bottom** [comment]
Sure, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/1109/)

(with exotics)

https://ci.cfengine.com/job/pr-pipeline/1109/

File: libpromises/files_hashes.c [hunk 1]
```diff
@@ -53,12 +53,21 @@ void HashFile(const char *filename, unsigned char digest[EVP_MAX_MD_SIZE + 1], H
...
+            (int) type);
+        fclose(file);
+        return;
+    }
 
     EVP_MD_CTX *context = EVP_MD_CTX_new();
     if (context == NULL)
     {
         Log(LOG_LEVEL_ERR, "Failed to allocate openssl hashing context");
+        fclose(file);
```
[the remark this answers is not in the document]
Comment (olehermanse) [reply libpromises/files_hashes.c]: It is, unwrap the UI below.

**Comment by olehermanse** [comment]
@craigcomstock About right. It is impossible to know exactly what went wrong from 1 backtrace, but I fixed the obvious programming mistakes.

File: libpromises/files_hashes.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (karlhto) [reply libpromises/files_hashes.c]: Must have overlooked it, because I looked for it and didn't find it. 👍

**Pull request merged**